test/test_quantizer.py

```diff
@@ -222,3 +225,110 @@ def forward(self, x):
         self.assertTrue('z' in eagerDict and 'z.1' in activationDict)
         self.assertAlmostEqual(eagerDict["z"][0], activationDict["z.1"][0], places=15)
         self.assertAlmostEqual(eagerDict["z"][1], activationDict["z.1"][1], places=15)
+
+    def test_compare_qparam_eager_script_default(self):
+        class Observer(torch.nn.Module):
+            __annotations__ = {'scale' : Optional[torch.Tensor], 'zero_point': Optional[torch.Tensor]}
+            def __init__(self):
+                super(Observer, self).__init__()
+                self.dtype = torch.quint8
+                self.qscheme = torch.per_tensor_affine
+                self.scale, self.zero_point = None, None
+
+            def forward(self, x):
+                self.scale = torch.tensor([2.0])
+                self.zero_point = torch.tensor([3])
+                return x
+
+            @torch.jit.export
+            def calculate_qparams(self):
+                return self.scale, self.zero_point
+
+        class WeightObserver(Observer):
+            def __init__(self):
+                super(WeightObserver, self).__init__()
+                self.dtype = torch.qint8
+
+        class TestM(nn.Module):
+            def __init__(self, qconfig):
+                super(TestM, self).__init__()
+                self.conv = nn.Conv2d(3, 1, 3).float()
+                self.conv.weight.data.fill_(1.0)
+                self.conv.bias.data.fill_(0.01)
+                self.qconfig = qconfig
+                self.quant = QuantStub()
+                self.dequant = DeQuantStub()
+
+            def forward(self, x):
+                return self.dequant(self.conv(self.quant(x)))
+
+        class TestScriptM(torch.jit.ScriptModule):
+            def __init__(self, init_weight=None):
+                super(TestScriptM, self).__init__()
+                self.conv = nn.Conv2d(3, 1, 3).float()
+                self.conv.bias.data.fill_(0.01)
+
+            @torch.jit.script_method
+            def forward(self, x):
+                y = self.conv(x)
+                return y
+
+        # Test Data
+        data = [(torch.randn(10, 3, 10, 10, dtype=torch.float), 1)]
+
+        # Eager mode
+        fake_qconfig = QConfig(activation=Observer, weight=WeightObserver)
+        eager_module = TestM(fake_qconfig)
+        script_module = TestScriptM()
+        script_module.conv.weight = torch.nn.Parameter(eager_module.conv.weight.detach())
+        quantized_eager_module = quantize(eager_module, default_eval_fn, data)
+        torch._C._jit_set_inline_everything_mode(False)
```

Comment:
It's better to use the corresponding decorator on a function to make sure we don't affect other tests.

Reply:
we actually need to inline for now and don't inline will be supported later, this line has no effect right now since it is after creation of ScriptModule